pkg/server/server.go

```diff
@@ -19,21 +19,29 @@ import (
 	"github.com/open-sauced/pizza/oven/pkg/providers"
 )
 
+// Config provides the configuration set on server startup
+// - Never Evict Repos: Repos that are preserved in cache regardless of LRU policy
+type Config struct {
+	NeverEvictRepos []string `yaml:"never-evict-repos"`
+}
+
 // PizzaOvenServer provides a leveled logger for use during serving requests
 // and a PizzaOvenDbHanlder for accessing a sql pool of connections.
 type PizzaOvenServer struct {
 	Logger           *zap.SugaredLogger
 	PizzaOven        *database.PizzaOvenDbHandler
 	PizzaGitProvider providers.GitRepoProvider
+	Config           *Config
 }
 
 // NewPizzaOvenServer returns a PizzaOvenServer with a new leveled logger
 // which uses the provided PizzaOvenHandler for db connections
-func NewPizzaOvenServer(dbHandler *database.PizzaOvenDbHandler, provider providers.GitRepoProvider, sugarLogger *zap.SugaredLogger) *PizzaOvenServer {
+func NewPizzaOvenServer(dbHandler *database.PizzaOvenDbHandler, provider providers.GitRepoProvider, sugarLogger *zap.SugaredLogger, config *Config) *PizzaOvenServer {
```

Comment:
This probably actually needs to be passed into the `NewLRUCacheGitRepoProvider` function in order for the cache eviction to not evict these configured repos.

Reply:
Addressed this on the latest commit, now config is drilled to the ```NewLRUCacheGitRepoProvider``` in order to be used for the cache.

```go
func NewLRUCacheGitRepoProvider(cacheDir string, minFreeDisk uint64, l *zap.SugaredLogger, config *Config) (GitRepoProvider, error) {
	cache, err := cache.NewGitRepoLRUCache(cacheDir, minFreeDisk)
	if err != nil {
		return nil, fmt.Errorf("could not initialize a new LRU cache: %s", err.Error())
	@@ -30,6 +37,7 @@ func NewLRUCacheGitRepoProvider(cacheDir string, minFreeDisk uint64, l *zap.Suga
	return &LRUCacheGitRepoProvider{
		logger:   l,
		LRUCache: cache,
		Config:   config,
	}, nil
}
```